### PR TITLE
Propagate cleared shell elements correctly

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
When clearing out shell collections the removal of those elements wasn't being propagated out correctly

### Platforms Affected ### 
- Core/XAML (all platforms)

### Testing Procedure ###
- unit tests included

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/Shell/ShellSectionCollection.cs [hunk 1]
```diff
@@ -119,7 +123,13 @@ void CheckVisibility(ShellSection section)
 
 		public void Add(ShellSection item) => Inner.Add(item);
 
-		public void Clear() => Inner.Clear();
+		public void Clear()
```
Comment (felipebaltazar) [inline Xamarin.Forms.Core/Shell/ShellSectionCollection.cs]: This logic is duplicated in `ShellItemCollection.cs`,  maybe we can create a base class with this code (and remove logic)
What do you think? 😊

Comment (PureWeen) [reply Xamarin.Forms.Core/Shell/ShellSectionCollection.cs]: It's probably about that time :-)
But I'll leave that for the next PR for now

There are type specifics that will probably need a few generics and probably some generic interfaces as well.

All good stuff but something better left for a later branch for now

**Comment by PureWeen** [comment]
ios12 test failure unrelated

File: Xamarin.Forms.Core/Shell/ShellSectionCollection.cs (hunk 1 above)
Comment (felipebaltazar) [reply Xamarin.Forms.Core/Shell/ShellSectionCollection.cs]: Maybe I can help with this, I've created a suggestion for this change.
What do you think?

https://github.com/xamarin/Xamarin.Forms/pull/9626

**Pull request merged**